src/lib.rs

```diff
@@ -17,46 +17,171 @@
 mod monerod;
 mod wallet;
 
+use anyhow::Result;
+use rand::Rng;
 use serde::{Deserialize, Serialize};
-
-const MONERO_WALLET_RPC_PORT: u16 = 28083; // Next available port after monerod's ports.
-const MONEROD_RPC_PORT: u16 = 28081; // Default testnet port.
-
-/// RPC client for monerod and monero-wallet-rpc.
+use std::time::Duration;
+use testcontainers::{
+    clients,
+    core::Port,
+    images,
+    images::generic::{GenericImage, Stream, WaitFor},
+    Container, Docker, Image,
+};
+
+/// When creating an account for Alice fund it with this amount of moneroj.
+const INITIAL_FUNDS_ALICE: u64 = 1000;
+/// How often we mine a block.
+const BLOCK_TIME_SECS: u64 = 1;
+/// Poll interval when checking if the wallet has synced with monerod.
+const WAIT_WALLET_SYNC_MILLIS: u64 = 1000;
+
+/// RPC client for `monerod` and `monero-wallet-rpc`.
 #[derive(Debug)]
-pub struct Client {
+pub struct Client<'c> {
+    container: Container<'c, clients::Cli, GenericImage>,
```

Comment:
It is a bit odd that this struct is called _Client_ if it hosts the container that runs the actual node. Seems like it might be worth separating out an actual node and let this struct host the two clients.

Reply:
How about just renaming it? If we pull the container out its just another thing that users have to instantiate and keep around, they already have to create the docker client.